src/coreclr/inc/nibblemapmacros.h

```diff
@@ -32,23 +39,30 @@
 # define CODE_ALIGN             sizeof(DWORD)                                // 4 byte boundry
 # define LOG2_CODE_ALIGN        2
 #endif
-#define NIBBLE_MASK             0xf
+#define NIBBLE_MASK             0xfu
 #define NIBBLE_SIZE             4                                            // 4 bits
 #define LOG2_NIBBLE_SIZE        2
-#define NIBBLES_PER_DWORD       ((8*sizeof(DWORD)) >> LOG2_NIBBLE_SIZE)      // 8 (4-bit) nibbles per dword
+#define NIBBLES_PER_DWORD       (2 * sizeof(DWORD))                          // 8 (4-bit) nibbles per dword
 #define NIBBLES_PER_DWORD_MASK  (NIBBLES_PER_DWORD - 1)                      // 7
 #define LOG2_NIBBLES_PER_DWORD  3
 #define BYTES_PER_BUCKET        (NIBBLES_PER_DWORD * CODE_ALIGN)             // 32 bytes per bucket
 #define LOG2_BYTES_PER_BUCKET   (LOG2_CODE_ALIGN + LOG2_NIBBLES_PER_DWORD)   //  5 bits per bucket
 #define MASK_BYTES_PER_BUCKET   (BYTES_PER_BUCKET - 1)                       // 31
+#define BYTES_PER_DWORD         (NIBBLES_PER_DWORD * BYTES_PER_BUCKET)       // 256 bytes per dword
 #define HIGHEST_NIBBLE_BIT      (32 - NIBBLE_SIZE)                           // 28 (i.e 32 - 4)
 #define HIGHEST_NIBBLE_MASK     (NIBBLE_MASK << HIGHEST_NIBBLE_BIT)          // 0xf0000000
 
 #define ADDR2POS(x)                      ((x) >> LOG2_BYTES_PER_BUCKET)
 #define ADDR2OFFS(x)            (DWORD)  ((((x) & MASK_BYTES_PER_BUCKET) >> LOG2_CODE_ALIGN) + 1)
 #define POSOFF2ADDR(pos, of)    (size_t) (((pos) << LOG2_BYTES_PER_BUCKET) + (((of) - 1) << LOG2_CODE_ALIGN))
-#define HEAP2MAPSIZE(x)                  (((x) / (BYTES_PER_BUCKET * NIBBLES_PER_DWORD)) * CODE_ALIGN)
+#define HEAP2MAPSIZE(x)                  (((x) / (BYTES_PER_DWORD) + 1) * sizeof(DWORD))
 #define POS2SHIFTCOUNT(x)       (DWORD)  (HIGHEST_NIBBLE_BIT - (((x) & NIBBLES_PER_DWORD_MASK) << LOG2_NIBBLE_SIZE))
 #define POS2MASK(x)             (DWORD) ~(HIGHEST_NIBBLE_MASK >> (((x) & NIBBLES_PER_DWORD_MASK) << LOG2_NIBBLE_SIZE))
 
+#define GETNIBBLE(dword, idx)   (DWORD)  ((dword >> POS2SHIFTCOUNT(idx)) & NIBBLE_MASK)
+#define ISPOINTER(dword)                 (GETNIBBLE(dword, NIBBLES_PER_DWORD - 1) > 8)
+#define ENCODEPOINTER(ptr)      (DWORD)  ((ptr & ~NIBBLE_MASK) + (((ptr & NIBBLE_MASK) >> 2) + 9))
+#define DECODEPOINTER(dword)    (size_t) ((dword & ~NIBBLE_MASK) + (((dword & NIBBLE_MASK) - 9) << 2))
```

Comment:
Do these really need to be macros?  Could they be static inline functions instead?